src/index.js

```diff
@@ -83,6 +87,13 @@ showBrowserWarning().then(function start() {
     onDone({ positions, barys, randoms, buildings, buildingIdxToMetadataList }) {
       loader.render(1)
       buffers.update({ positions, barys, randoms, buildings }, stateTransitioner.getStateIndexes())
+      curPositionsLoaded = positions.length / 3
+      
+      // stateTransitioner.updateLoadingState(latest.buildingIdxToMetadataList)
+      buffers.update({ positions, barys, randoms, buildings }, stateTransitioner.getStateIndexes())
+      const attrs = buffers.getAttributes()
+      renderBuildings = createBuildingsRenderer(regl, positions, barys, randoms, stateTransitioner.getStateIndexes(), settings)
```

Comment:
Ohhhhh - this is where I misread. I missed that you're passing these directly to `createBuildingsRenderer` and line 93 & 94 here are dead. Damn - so maybe there's a bug in the offset/stride stuff, like you said. 🤔